lib/stripe.ex

```diff
@@ -332,8 +332,11 @@ defmodule Stripe do
   end
 
   @spec handle_response(http_success | http_failure) :: {:ok, map} | {:error, api_error_struct}
-  defp handle_response({:ok, status, _headers, body}) when status in 200..299 do
-    decoded_body = Poison.decode!(body)
+  defp handle_response({:ok, status, headers, body}) when status in 200..299 do
+    decoded_body = 
+      body
+      |> Poison.decode!
+      |> decompress_body(headers)
```

Comment:
@tlvenn Should this be before `Poison.decode!/2`? Otherwise won't it try to decode to JSON the compressed body?